src/item.cpp

```diff
@@ -5963,6 +5982,11 @@ bool item::is_money() const
     return ammo_types().count( ammotype( "money" ) );
 }
 
+bool item::is_software() const
+{
+    return type->default_container->str() == "usb_drive";
```

Comment:
```suggestion
    cata::optional<itype_id> optional_itype_usb_drive( "usb_drive" );
    return type->default_container == optional_itype_usb_drive;
}

bool item::is_software_storage() const
{
    const itype_id itype_usb_drive( "usb_drive" );
    return typeId() == itype_usb_drive;
```
Change `itype_id` check. Add `item::is_software_storage()`.